### PR TITLE
feat: update k6 dependency and go version

### PR DESCRIPTION
**Pull request by fgouteroux**
Hello,

Since **k6 version v0.42**  this extension **xk6-output-prometheus-remot**e is available within k6 as an experimental module.
I made this PR to update the k6 dependency version to v0.45 and go version to 1.19.

So we can do this:
```
k6 run load-testing-remote-write.js -o experimental-prometheus-rw

### CODE REVIEW

**Review by javaducky — CHANGES_REQUESTED** [review]
Hi @fgouteroux! Thank you for the update!

It looks like the `bench_test.go` no longer compiles with these changes. You can see this issue by running `go test -cover -race ./...`. Please correct the compilation issues and we'll reevaluate.

**Comment by javaducky** [comment]
> So we can do this:
> 
> ```
> k6 run load-testing-remote-write.js -o experimental-prometheus-rw
> ```

@fgouteroux , there may be some confusion with this particular extension. _This_ extension (`xk6-client-prometheus-remote`) is for testing Prometheus remote write endpoints directly, i.e. testing Prometheus, DataDog, New Relic installations.

The [xk6-output-prometheus-remote](https://github.com/grafana/xk6-output-prometheus-remote) is the extension allowing k6 test metrics to be output to a Prometheus backend. It is this output extension that is invoked as an experimental module using the `-o experimental-prometheus-rw` argument to k6.

I hope this clears up any confusion on these similarly named extensions.

**Comment by javaducky** [comment]
This issue would address #29.

**Comment by fgouteroux** [comment]
> > So we can do this:
> 
> > 
> 
> > ```
> 
> > k6 run load-testing-remote-write.js -o experimental-prometheus-rw
> 
> > ```
> 
> 
> 
> @fgouteroux , there may be some confusion with this particular extension. _This_ extension (`xk6-client-prometheus-remote`) is for testing Prometheus remote write endpoints directly, i.e. testing Prometheus, DataDog, New Relic installations.
> 
> 
> 
> The [xk6-output-prometheus-remote](https://github.com/grafana/xk6-output-prometheus-remote) is the extension allowing k6 test metrics to be output to a Prometheus backend. It is this output extension that is invoked as an experimental module using the `-o experimental-prometheus-rw` argument to k6.
> 
> 
> 
> I hope this clears up any confusion on these similarly named extensions.
> 
> 

Yes I know and this is what I want. Testing my mimir clusters and output the result into it. I use the grafana k6 dashboard as well and we can track our tests result.

**Comment by fgouteroux** [comment]
> Hi @fgouteroux! Thank you for the update!
> 
> It looks like the `bench_test.go` no longer compiles with these changes. You can see this issue by running `go test -cover -race ./...`. Please correct the compilation issues and we'll reevaluate.

@javaducky  fixed

File: go.mod [hunk 1]
```diff
@@ -1,47 +1,48 @@
...
-go 1.18
+go 1.19
 
 require (
-	github.com/dop251/goja v0.0.0-20221003171542-5ea1285e6c91
-	github.com/golang/protobuf v1.5.2
+	github.com/dop251/goja v0.0.0-20230531210528-d7324b2d74f7
+	github.com/golang/protobuf v1.5.3
 	github.com/golang/snappy v0.0.4
 	github.com/oxtoacart/bpool v0.0.0-20190530202638-03653db5a59c
```
Comment (javaducky) [inline go.mod:9]: Please run `go mod tidy`. The requirement for `oxtoacart/bpool` is no longer necessary.

**Review by javaducky — CHANGES_REQUESTED** [review]

**Review by javaducky — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**